scripts/o3de/tests/test_cmake.py

```diff
@@ -205,10 +206,71 @@ def get_gems_json_data_by_name(engine_path:pathlib.Path = None,
 
         assert result == expected_result
         assert resolved_gem_paths == expected_gem_paths
+
+
+class TestUpdateCMakePresetsJson:
+    @pytest.mark.parametrize(
+        "preset_path, engine_name, engine_version, preset_json_content, expected_result",
+        [
+            # This should create the cmakepresets file and add the registered 'o3de' engine to it
+            pytest.param('user/CMakePresets.json', 'o3de', '', '', cmake.UpdatePresetResult.EnginePathAdded),
+            # This should create the cmake-presets file and add the registered 'o3de==1.0.0' engine to it
+            pytest.param('user/CMakePresets.json', 'o3de', '1.0.0', '',cmake.UpdatePresetResult.EnginePathAdded),
+            # This should fail to create the cmake-presets file as 'o3de==2.0.0' is not registered
+            pytest.param('user/CMakePresets.json', 'o3de', '2.0.0', '', cmake.UpdatePresetResult.Error),
+            # This should fail to create the cmake-presets file as 'o3de-not-registered' is not registered
+            pytest.param('user/CMakePresets.json', 'o3de-not-registered', '', '', cmake.UpdatePresetResult.Error),
+            # This should fail to create the cmake-presets file as the file path does not exist
+            pytest.param('', 'o3de', '1.0.0', '', cmake.UpdatePresetResult.Error),
+            # This should update an existing cmake-presets file
+            pytest.param('user/CMakePresets.json', 'o3de', '',
+                          string.Template(cmake.TEMPLATE_CMAKE_PRESETS_INCLUDE_JSON).safe_substitute(CMakePresetsInclude='<other-engine-root>/CMakePresets.json'),
+                          cmake.UpdatePresetResult.EnginePathAdded),
+            # This should update an existing cmake-presets file
+            pytest.param('user/CMakePresets.json', 'o3de', '',
+                          string.Template(cmake.TEMPLATE_CMAKE_PRESETS_INCLUDE_JSON).safe_substitute(CMakePresetsInclude='<engine-root>/CMakePresets.json'),
+                          cmake.UpdatePresetResult.EnginePathAlreadyIncluded)
+        ]
+    )
+    def test_update_cmake_presets_for_project(self, preset_path, engine_name, engine_version,
+                                              preset_json_content, expected_result):
+        class StringBufferIOWrapper(io.StringIO):
+            def __init__(self):
+                nonlocal preset_json_content
+                super().__init__(preset_json_content)
+            def __enter__(self):
+                return super().__enter__()
+            def __exit__(self, exc_type, exc_val, exc_tb):
+                nonlocal preset_json_content
+                preset_json_content = super().getvalue()
+                super().__exit__(exc_tb, exc_val, exc_tb)
+
+        def preset_json_open(mode):
+            if preset_path:
+                return StringBufferIOWrapper()
+            raise OSError('Cannot Open file "{preset_path}"')
```

Comment:
I wonder if it's worth including `preset_path` as a value here because if `preset_path` is non-empty won't it not reach this line?

Reply:
Well the message itself isn't actually import.
The only thing need was to raise an OSError so that the UnitTest replicates being unable to open the presets file.
It could be an OSError with no message.